cmd/agent/subcommands/secret/command.go

```diff
@@ -32,55 +31,74 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 	cliParams := &cliParams{
 		GlobalParams: globalParams,
 	}
+
 	secretInfoCommand := &cobra.Command{
 		Use:   "secret",
-		Short: "Print information about decrypted secrets in configuration.",
+		Short: "Display information secrets in configuration.",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return fxutil.OneShot(showSecretInfo,
+				fx.Supply(command.GetDefaultCoreBundleParams(cliParams.GlobalParams)),
+				core.Bundle(),
+			)
+		},
+	}
+	secretRefreshCommand := &cobra.Command{
+		Use:   "refresh",
+		Short: "Refresh secrets in configuration.",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return fxutil.OneShot(secret,
-				fx.Supply(cliParams),
+			return fxutil.OneShot(secretRefresh,
 				fx.Supply(command.GetDefaultCoreBundleParams(cliParams.GlobalParams)),
 				core.Bundle(),
 			)
 		},
 	}
+	secretInfoCommand.AddCommand(secretRefreshCommand)
 
 	return []*cobra.Command{secretInfoCommand}
 }
 
-func secret(_ log.Component, config config.Component, _ *cliParams) error {
-	if err := util.SetAuthToken(); err != nil {
-		fmt.Println(err)
-		return nil
+func showSecretInfo(config config.Component) error {
+	r, err := callAPIEndpoint("/agent/secrets", config)
+	if err != nil {
+		return err
 	}
+	fmt.Println(string(r))
+	return nil
+}
 
-	if err := showSecretInfo(config); err != nil {
-		fmt.Println(err)
-		return nil
+func secretRefresh(config config.Component) error {
+	r, err := callAPIEndpoint("/agent/secret/refresh", config)
+	if err != nil {
+		return err
 	}
+	fmt.Printf("Secrets refresh: %s\n", r)
 	return nil
 }
 
-func showSecretInfo(config config.Component) error {
+func callAPIEndpoint(apiEndpointPath string, config config.Component) ([]byte, error) {
+	if err := util.SetAuthToken(); err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
 	c := util.GetClient(false)
 	ipcAddress, err := pkgconfig.GetIPCAddress()
```

Comment:
Could we use the `config.Component` to get this value?

Reply:
Currently the config.Component is very minimal, and doesn't provide a way to do this. I think it would be better if it did, but there's lots of places throughout the subcommands that are using code exactly like this. Would probably be better as its own stand-alone change that refactors all of these places.